contracts/lbp/LBPWrapper.sol

```diff
@@ -41,96 +43,81 @@ contract LBPWrapper {
         owner = _newOwner;
     }
 
-    /**
-     * @dev                       initialize lbp wrapper contract
-     * @param _LBPFactory         LBP factory address
-     * @param _swapFeePercentage  swap fee percentage
-     */
-    function initialize(address _LBPFactory, uint256 _swapFeePercentage)
-        public
-    {
-        owner = msg.sender;
-        LBPFactory = ILBPFactory(_LBPFactory);
-        swapFeePercentage = _swapFeePercentage;
-        isInitialized = true;
-    }
-
     /**
      * @dev                        initialize lbp wrapper contract
+     * @param _LBPFactory          LBP factory address
      * @param _name                LBP name
      * @param _symbol              LBP symbol
      * @param _tokens              array of tokens sorted for the LBP
-     * @param _amounts             array of initial amounts used to join pool
      * @param _weights             array of start weights for respective tokens
      * @param _swapEnabledOnStart  enable or disable swap
      * @param _startTime           start time
      * @param _endTime             end time
      * @param _endWeights          array of end weights for respective tokens
-     * @param _admin               address of admin/owner of LBP
-     * @param _userData            extra data required by LBP
      */
-    function deployLbpFromFactory(
+    function initializeLBP(
+        address _LBPFactory,
         string memory _name,
         string memory _symbol,
         IERC20[] memory _tokens,
-        uint256[] memory _amounts,
         uint256[] memory _weights,
         bool _swapEnabledOnStart,
         uint256 _startTime,
         uint256 _endTime,
-        uint256[] memory _endWeights,
-        address _admin,
-        bytes memory _userData
-    ) public onlyOwner returns (address) {
-        // to handle stack overflow
-        {
-            address vault = address(LBPFactory.getVault());
-            for (uint8 i; i < _tokens.length; i++) {
-                IERC20(_tokens[i]).approve(vault, _amounts[i]);
-            }
-        }
+        uint256[] memory _endWeights
+    ) public returns (address) {
```

Comment:
I understand this function is called by the LBPWrapperFactory upon deploying. Could we change the function to `external`?

Reply:
yes, would be great. many functions might need to be marked as external.